openedx/features/learner_analytics/views.py

```diff
@@ -97,3 +109,84 @@ def get_schedule(self, request, course_key):
             if all_blocks['blocks'][block].get('graded', False):
                 graded_blocks[block] = all_blocks['blocks'][block]
         return json.dumps(graded_blocks)
+
+    def get_weekly_course_activities(self, course_key):
+        """
+        Get the count of any course activity from previous 7 days
+
+        Args:
+            course_key: CourseKey
+        """
+        cache_key = 'learner_analytics_{course_key}_weekly_activities'.format(course_key=course_key)
+        activities = cache.get(cache_key)
+
+        if not activities:
+            log.info('Weekly course activities for course {course_key} was not cached - fetching from Analytics API'
+                     .format(course_key=course_key))
+            activities = self.analytics_client.courses(course_key).activity()
+
+            seconds_to_cache_expiration = LearnerAnalyticsView.seconds_to_cache_expiration()
+            log.info('Weekly course activities cache expiration: {seconds_to_cache_expiration}'
+                     .format(seconds_to_cache_expiration=seconds_to_cache_expiration))
+            cache.set(cache_key, activities[0], seconds_to_cache_expiration)
+            activities = activities[0]
+
+        return activities['any']
+
+    @staticmethod
+    def consecutive_active_days_for_user(username, course_key):
+        """
+        Get the most recent count of consecutive days that a use has performed a course activity
+
+        Args:
+            username: Username
+            course_key: CourseKey
+        """
+        cache_key = 'learner_dashboard_{username}_{course_key}_engagement_timeline'\
+            .format(username=username, course_key=course_key)
+        timeline = cache.get(cache_key)
+
+        if not timeline:
+            log.info('Engagement timeline for course {course_key} was not cached - fetching from Analytics API'
+                     .format(course_key=course_key))
+
+            # TODO: @jaebradley replace this once the Analytics client has an engagement timeline method
```

Comment:
[The Analytics Client PR that does this](https://github.com/edx/edx-analytics-data-api-client/pull/37)